### PR TITLE
Add note clarifying feature availability

### PR DESCRIPTION
**Pull request by Maximisch**
### Why:

Minor adjustment to make feature availability more clear for customers.
People are currently frustrated that this feature cannot be activated for private repositories only (see e.g. [here](https://github.com/github/feedback/discussions/4264#discussioncomment-1465140)), and this minor note should help avoid future frustration.

### What's being changed:

Added the sentence "Issue forms can currently be enabled on private repositories only." to the note box on issue forms.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@Maximisch 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: data/reusables/community/issue-forms-beta.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% note %}
 
-**Note:** Issue forms are currently in beta and subject to change.
+**Note:** Issue forms are currently in beta and subject to change. Issue forms can currently be enabled on private repositories only.
```
Comment (lecoursen) [inline data/reusables/community/issue-forms-beta.md:3]:
```suggestion
**Note:** Issue forms are currently in beta for public repositories on {% data variables.product.prodname_dotcom_the_website %} only.
```

**Review by lecoursen — COMMENTED** [review]
👋🏻 @Maximisch, I agree that this would be helpful to clarify! While reviewing this I noticed that "[Syntax for issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)" seems to be repeating this beta information twice:

image

Adding the new information to the reusable here would lead to even more duplication. 😭 Would you mind removing that `product` property from the front matter of that article in this PR as well? Thanks!

**Comment by Maximisch** [comment]
@lecoursen I removed the product tag as you suggested! 👍

**Review by lecoursen — APPROVED** [review]

**Pull request merged**